### ISSUE TITLE
Make Popper example more accessible

### ISSUE THREAD

**Issue by NicholasBoll** [Workday/canvas-kit#283]
## 🐛 Bug Report

Popper example isn't as accessible as it could be

## To Reproduce

Navigate to https://workday.github.io/canvas-kit/?path=/story/common--popper and click "Toggle Popper"

## Expected Behavior

* The first `h3` tag should have an `id` attribute with a value like `'my-modal'`
* The `[role=dialog]` element should have a `aria-labelledby` with a value like `'my-modal'` (to associate the dialog being labelled by the heading)
* The second `h3` tag should be reduced to a `p` tag

## Actual Results

* `[role=dialog]` is not labelled by anything
* Second `h3` tag doesn't provide any semantic meaning to the Popper element

## Link to repl or repo (highly encouraged)

https://workday.github.io/canvas-kit/?path=/story/common--popper

**Comment by jpante** [comment]
The Popper story is an example of leveraging this utility. Closing this as future work will add more thoughtful stories and documentation for other popup types in context.
